tcp-echo.go

```diff
@@ -89,6 +126,52 @@ func listenAndHandle(listener net.Listener, message string) {
 	}
 }
 
+func listenPacketAndHandle(conn net.PacketConn, message string) {
+	for {
+		buf := make([]byte, 1024)
+		size, addr, err := conn.ReadFrom(buf)
+		if err != nil {
+			log.Println("Read error:", err)
+			continue
+		}
+		_, err = conn.WriteTo([]byte(message), addr)
+		if err != nil {
+			log.Println("Message write error:", err)
+			continue
+		}
+		data := buf[:size]
+		log.Println("Received Raw UDP Data:", data)
+		log.Printf("Received UDP Data (converted to string): %s", data)
+		_, err = conn.WriteTo(data, addr)
+		if err != nil {
+			log.Println("Echo write error:", err)
+			continue
+		}
+		continue
+	}
+
+}
+
+func generateHTTPHandler(message string) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
+		callUuidV4, _ := uuid.NewUUID()
+		clientId := callUuidV4.String()
+
+		log.Println(clientId + " - HTTP connection open.")
+		defer log.Println(clientId + " - HTTP connection closed.")
+
+		body, err := io.ReadAll(r.Body)
+		if err != nil {
+			log.Println(clientId+" - HTTP body read error: ", err)
```

Comment:
The message string, which is honestly what I'm more interested in most of the time. The "Hello, Kubernetes!" response with the pod info is more useful in doc examples or tests to confirm that we successfully proxied a response, and from where.

I don't actually have much use for the body echo (maybe to demo request-transformer-advanced?), so if it somehow fails, just logging it should suffice. It's a toy app and doesn't need the world's most robust error handling, since there's not really any useful way for it to recover.

Reply:
All I'm trying to suggest here is that this might cause a lot of frustration when e.g. reading the body fails the and server returns something that's not entirely expected.

Imagine using this in tests for that example. You write asserts, expect a thing and then without communicating through status code the server doesn't return that and one needs to go through pod logs which, oh wait yes, they are on CI which probably didn't store them. You can see where I'm going with this.

The cost of adding these error handling statements is negligible IMHO to the cost of future frustration **if** someone ever decides to use it for such use case as mentioned above.

WDYT?